src/main/resources/init-script-gradle.ftl

```diff
@@ -132,8 +133,16 @@ def findProjectOutputFile(Project project, String outputDirectoryPath) {
     File outputDirectory = createTaskOutputDirectory(outputDirectoryPath)
     String name = project.toString()
 
+    int depthCount = 0
+    for(char c: name.toCharArray()) {
+        if (c == ':') {
+            depthCount++
+        }
+    }
+    String depth = String.valueOf(depthCount)
+
     String nameForFile = name?.replaceAll(/[^\p{IsAlphabetic}\p{Digit}]/, "_")
-    File outputFile = new File(outputDirectory, "${nameForFile}_dependencyGraph.txt")
+    File outputFile = new File(outputDirectory, "${nameForFile}_depth${depth}_dependencyGraph.txt")
```

Comment:
Currently, the dependency graphs are parsed in random order and there could be a case where a child is parsed before its parent making the gradle rich version declaration incorrect in such cases since we would have not get that information yet. 
With this we are adding depth value in the file name based on the number of colons`:` the project name has. This is mentioned in the Gradle documentation as the way of mentioning the projects. ( https://docs.gradle.org/current/userguide/intro_multi_project_builds.html ). So if there are three colons, then the depth is 3 and there are two parent projects above that particular project ( Eg: ':api:services:webservice')